profiles/common/modules/custom/nexteuropa_lastupdate/nexteuropa_lastupdate.module

```diff
@@ -0,0 +1,192 @@
+          }
+        }
+      }
+      if (!isset($block_content)) {
+        $block_content = '';
+      }
+      $block['content'] = $block_content;
+      break;
+  }
+  return $block;
+}
+
+/**
+ * Implements hook_entity_info_alter().
+ */
+function nexteuropa_lastupdate_entity_info_alter(&$entity_info) {
+  foreach (_nexteuropa_lastupdate_supported_core_entities() as $type => $callback) {
+    $entity_info[$type]['last update callback'] = $callback;
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function nexteuropa_lastupdate_theme($existing, $type, $theme, $path) {
+  return array(
+    'nexteuropa_last_update_block' => array(
+      'template' => 'nexteuropa-last-update-block',
+      'path' => $path . '/theme',
+    ),
+  );
+}
+
+/**
+ * Define last update callbacks for default entities.
+ *
+ * @return array $last_update_callbacks.
+ *   Array of callbacks keyed by entity name.
+ */
+function _nexteuropa_lastupdate_supported_core_entities() {
+  $last_update_callbacks = array();
+  // Core entity types.
+  if (module_exists('comment')) {
+    $last_update_callbacks['comment'] = '_nexteuropa_lastupdate_comment_callback';
+  }
+  $last_update_callbacks['user'] = '_nexteuropa_lastupdate_user_callback';
+  $last_update_callbacks['node'] = '_nexteuropa_lastupdate_node_callback';
+  $last_update_callbacks['file'] = '_nexteuropa_lastupdate_file_callback';
+
+  return $last_update_callbacks;
+}
+
+/**
+ * Return the entity that is being loaded in the page (if any).
+ *
+ * @return mixed
+ *   On failure false.
+ *   On success the loaded entity object.
+ */
+function _nexteuropa_lastupdate_menu_get_any_entity() {
+  // Figure out how this entity is loaded.
+  $entity_type = FALSE;
+  $item = menu_get_item();
+
+  // Non-entities may not have load functions.
+  if (is_array($item['load_functions'])) {
+    $vals = array_values($item['load_functions']);
+    $load_function = $vals[0];
+    $arg_position = array_search($load_function, $item['load_functions']);
+
+    // Compare to entity types.
+    $entity_info = entity_get_info();
+    foreach ($entity_info as $type => $info) {
+      if ($info['load hook'] == $load_function) {
+        $entity_type = $type;
+      }
+    }
+  }
+
+  // Return the entity.
+  if ($entity_type && $entity = menu_get_object($entity_type, $arg_position)) {
+    if (is_object($entity)) {
+      $entity->entity_type = $entity_type;
+      return $entity;
+    }
+    else {
+      return FALSE;
+    }
+  }
+  else {
+    return FALSE;
+  }
+}
+
+
+/**
+ * Callback to return the last published date of a node.
+ *
+ * @param object $node
+ *   The node to get the last published date from.
+ *
+ * @return array
+ *   An array containing the label and date to be rendered.
+ */
+function _nexteuropa_lastupdate_node_callback($node) {
+  if ($node->status) {
+    return array(
+      'label' => t('Last published'),
+      'date' => date('d/m/Y', $node->changed),
```

Comment:
One QA rule is to use as much as possible Drupal functions. In this case to format it, we have [format_date](https://api.drupal.org/api/drupal/includes!common.inc/function/format_date/7.x) that should `replace date()`.

Furthermore, we should not enforce the date format like it is done. We should use those supplied by Drupal en existing in Drupal: 'long', short or medium.
Up to you to determine which one to choose.